### PR TITLE
Disable all colors in Buildpacks’s output when not in a terminal

### PR DESCRIPTION
**Pull request by dgageot**
+ Make sure the output from pack and from buildpacks goes to stdout
 + Make sure all colors are disabled when not printing to a terminal

### CODE REVIEW

File: pkg/skaffold/build/buildpacks/logger.go [hunk 1]
```diff
@@ -30,12 +33,27 @@ type logger struct {
...
 
 func NewLogger(out io.Writer) logging.Logger {
+	// If out is not a terminal, let's make sure no colors are printed.
+	if _, isTerm := util.IsTerminal(out); !isTerm {
+		out = colorable.NewNonColorable(out)
+	}
+
+	l := logrus.New()
+	l.SetOutput(out)
+	l.SetFormatter(new(plainFormatter))
```
Comment (briandealwis) [inline pkg/skaffold/build/buildpacks/logger.go]: I'd appreciate a comment here as to why this formatter is required

File: pkg/skaffold/build/buildpacks/lifecycle.go [hunk 2]
```diff
@@ -85,12 +83,6 @@ func (b *Builder) build(ctx context.Context, out io.Writer, a *latest.Artifact,
 }
 
 func runPackBuild(ctx context.Context, out io.Writer, localDocker docker.LocalDaemon, opts pack.BuildOptions) error {
-	// If out is not a terminal, let's make sure pack doesn't output with colors.
-	if _, isTerm := util.IsTerminal(out); !isTerm {
-		// pack uses heroku/color under the hood.
-		color.Disable(true)
```
Comment (briandealwis) [inline pkg/skaffold/build/buildpacks/lifecycle.go]: Oh, I guess a problem here is that we never re-enable color.

File: pkg/skaffold/build/buildpacks/logger.go (hunk 1 above)
Comment (dgageot) [reply pkg/skaffold/build/buildpacks/logger.go]: Added

File: pkg/skaffold/build/buildpacks/lifecycle.go (hunk 2 above)
Comment (dgageot) [reply pkg/skaffold/build/buildpacks/lifecycle.go]: Yes, the new code is more robust

**Pull request merged**